rewrite-javascript/build.gradle.kts

```diff
@@ -55,68 +56,64 @@ extensions.configure<NodeExtension> {
     nodeProjectDir.set(projectDir.resolve("rewrite"))
 }
 
-val npmTest = tasks.named("npm_test")
-npmTest.configure {
+val npmInstall = tasks.named("npmInstall")
+
+val npmTest = tasks.register<NpmTask>("npmTest") {
+    inputs.files(npmInstall)
+        .withPathSensitivity(PathSensitivity.RELATIVE)
     inputs.files(fileTree("rewrite") {
         include("*.json")
         include("jest.config.js")
-    })
+    }).withPathSensitivity(PathSensitivity.RELATIVE)
     inputs.files(fileTree("rewrite/src"))
+        .withPathSensitivity(PathSensitivity.RELATIVE)
     inputs.files(fileTree("rewrite/test"))
+        .withPathSensitivity(PathSensitivity.RELATIVE)
     outputs.files("rewrite/build/test-results/jest/junit.xml")
 
-    dependsOn(tasks.named("npmInstall"))
```

Comment:
I can see that we now instead have `inputs.files(npmInstall)`. But is that the same as also depending on the task?

Reply:
It is! As we can see [upstream](https://github.com/node-gradle/gradle-node-plugin/blob/main/src/main/kotlin/com/github/gradle/node/npm/task/NpmInstallTask.kt) the outputs of the `npmInstall` task include the `node_modules` directory which is exactly what we depend upon here for this task to execute properly.

Internally, to Gradle when inputs and outputs are tied together there are dependencies between the tasks included directly from that modeling.